src/timer.rs

```diff
@@ -234,6 +234,26 @@ macro_rules! hal {
                     self.tim.arr.write(|w| unsafe { w.bits(u32(arr)) });
                 }
 
+                pub fn set_timeout(&mut self, timeout: core::time::Duration) {
```

Comment:
time::Duration might surely be a thing to look into, however as of now we usually use our self defined type Milliseconds from time.rs, probably we could look into replacing it with time::Duration though.

However as of now I'd prefer to keep the code consistent unless someone has a good point against it.

Reply:
In fact I would not be surprised if Rust could optimize these conversion lots of the time.